### DISCUSSION TITLE
File Name Clears When Selecting Folder

### DISCUSSION THREAD

**Discussion by gabrielAtManticore** [yasirkula/UnitySimpleFileBrowser#48] [Ideas]
I just started using the file browser and noticed that when saving a file, if you click on any folder it clears the name of the file. This doesn't seem like the normal UX pattern on Windows or Mac. This can be changed by removing line 2190 of FileBrowser.cs where it says:
```
filenameInputField.text = string.Empty;
```
This fixes my immediate problem, but I don't know yet if this change would cause other issues. The name still updates when I click on a file, which is fine. So far so good.

**Comment by yasirkula** [comment]
It wouldn't look good if multiple files were previously selected. Filename input field would continue showing `"File1" "File2" "File3" ...`. But we can reset the input field text if only a single file was previously selected. What do you think:

```csharp
if( filenameContributingFileCount == 0 )
{
	// If multiple files were previously selected, clear the input field. If a single file was selected, preserve the filename
	if( filenameInputField.text.StartsWith( "\"" ) )
		filenameInputField.text = string.Empty;
}
```

**Comment by gabrielAtManticore** [comment]
My idea would be to approximate what's already established as a UX. I went
ahead and tested the multi-file case in Windows and it does not clear the
field in that case. That said, it's a very niche use case, so in my opinion
it could go either way.

**Reply by yasirkula** [reply]
When a lot of files are selected, only the first few files are included in the filename input field and the rest are replaced with `...` for performance reasons (imagine selecting all files in a directory and that the directory had hundreds of files, generating a string from these files' names and then updating the Text with that string would take unnecessarily long time). So the filename input field doesn't match the selection for large selections and I think it is best to clear the input field in that case.

But perhaps a better way would be to clear the input field only if it starts with `"` and ends with `...`.

**Comment by gabrielAtManticore** [comment]
Makes sense to me, you know all the edge cases much better!
